snuba/request/schema.py

```diff
@@ -125,3 +131,26 @@ def __generate_template_impl(self, schema) -> Any:
 
     def generate_template(self) -> Any:
         return self.__generate_template_impl(self.__composite_schema)
+
+
+SETTINGS_SCHEMA = {
+    HTTPRequestSettings: {
+        "type": "object",
+        "properties": {
+            # Never add FINAL to queries, enable sampling
+            "turbo": {"type": "boolean", "default": False},
+            # Force queries to hit the first shard replica, ensuring the query
+            # sees data that was written before the query. This burdens the
+            # first replica, so should only be used when absolutely necessary.
+            "consistent": {"type": "boolean", "default": False},
+            "debug": {"type": "boolean", "default": False},
+        },
```

Comment:
This is an extremely long-term thing (probably on the same time scale as the larger `Schema` abstraction mentioned in https://github.com/getsentry/snuba/pull/648#discussion_r360207274), but it'd be really cool if these JSON schema defaults were taken from the constructor defaults.

I think pipe dream is that the request and query data structures are defined as dataclasses, and the JSON schema is built from the dataclass definition — defaults and constraint validation and all that.